pkg/registry/core/namespace/strategy.go

```diff
@@ -88,6 +91,19 @@ func (namespaceStrategy) Validate(ctx context.Context, obj runtime.Object) field
 
 // Canonicalize normalizes the object after validation.
 func (namespaceStrategy) Canonicalize(obj runtime.Object) {
+	// Ensure the label matches the name for namespaces just created using GenerateName,
+	// where the final name wasn't available for defaulting to make this change.
+	// This code needs to be kept in sync with the implementation that exists
+	// in Namespace defaulting (pkg/apis/core/v1)
```

Comment:
I'd like to see this document the lifecycle more:

```
// Why this hook *and* defaults.go?
//
// CREATE:
// Defaulting and PrepareForCreate happen before generateName is completed 
// (i.e. the name is not yet known). Validation happens after generateName
// but should not modify objects. Canonicalize happens later, which makes 
// it the best hook for setting the label.
//
// UPDATE:
// Defaulting and Canonicalize will both trigger with the full name.
//
// GET:
// Only defaulting will be applied.
```

Reply:
added. Do you want me to add the same comment on defaults.go, or here in strategy is enough?